### PR TITLE
Use smaller tarballs, and a couple other minor changes

### PR DESCRIPTION
**Pull request by triallax**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/curl.rb [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
   desc "Get a file from an HTTP, HTTPS or FTP server"
   homepage "https://curl.se"
-  url "https://curl.se/download/curl-7.83.1.tar.bz2"
-  mirror "https://github.com/curl/curl/releases/download/curl-7_83_1/curl-7.83.1.tar.bz2"
-  mirror "http://fresh-center.net/linux/www/curl-7.83.1.tar.bz2"
-  mirror "http://fresh-center.net/linux/www/legacy/curl-7.83.1.tar.bz2"
-  sha256 "f539a36fb44a8260ec5d977e4e0dbdd2eee29ed90fcedaa9bc3c9f78a113bff0"
+  url "https://curl.se/download/curl-7.83.1.tar.xz"
+  mirror "https://github.com/curl/curl/releases/download/curl-7_83_1/curl-7.83.1.tar.xz"
+  mirror "http://fresh-center.net/linux/www/curl-7.83.1.tar.xz"
```
Comment (carlocab) [inline Formula/curl.rb:6]: Some formulae have HTTP mirrors in order to enable building them from source with an old version of `curl`. Using a `.xz` source tarball means that `xz` becomes a build dependency for these formulae, which we don't want.

File: Formula/gettext.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
 class Gettext < Formula
   desc "GNU internationalization (i18n) and localization (l10n) library"
   homepage "https://www.gnu.org/software/gettext/"
-  url "https://ftp.gnu.org/gnu/gettext/gettext-0.21.tar.gz"
-  mirror "https://ftpmirror.gnu.org/gettext/gettext-0.21.tar.gz"
-  mirror "http://ftp.gnu.org/gnu/gettext/gettext-0.21.tar.gz"
-  sha256 "c77d0da3102aec9c07f43671e60611ebff89a996ef159497ce8e59d075786b12"
+  url "https://ftp.gnu.org/gnu/gettext/gettext-0.21.tar.xz"
+  mirror "https://ftpmirror.gnu.org/gettext/gettext-0.21.tar.xz"
+  mirror "http://ftp.gnu.org/gnu/gettext/gettext-0.21.tar.xz"
```
Comment (carlocab) [inline Formula/gettext.rb:6]: Same here.

File: Formula/libunistring.rb [hunk 3]
```diff
@@ -1,10 +1,10 @@
 class Libunistring < Formula
   desc "C string library for manipulating Unicode strings"
   homepage "https://www.gnu.org/software/libunistring/"
-  url "https://ftp.gnu.org/gnu/libunistring/libunistring-1.0.tar.gz"
-  mirror "https://ftpmirror.gnu.org/libunistring/libunistring-1.0.tar.gz"
-  mirror "http://ftp.gnu.org/gnu/libunistring/libunistring-1.0.tar.gz"
-  sha256 "3c0184c0e492d7c208ce31d25dd1d2c58f0c3ed6cbbe032c5b248cddad318544"
+  url "https://ftp.gnu.org/gnu/libunistring/libunistring-1.0.tar.xz"
+  mirror "https://ftpmirror.gnu.org/libunistring/libunistring-1.0.tar.xz"
+  mirror "http://ftp.gnu.org/gnu/libunistring/libunistring-1.0.tar.xz"
```
Comment (carlocab) [inline Formula/libunistring.rb:6]: Same comment.

File: Formula/xz.rb [hunk 4]
```diff
@@ -3,9 +3,9 @@
 class Xz < Formula
   desc "General-purpose data compression with high compression ratio"
   homepage "https://tukaani.org/xz/"
-  url "https://downloads.sourceforge.net/project/lzmautils/xz-5.2.5.tar.gz"
-  mirror "https://tukaani.org/xz/xz-5.2.5.tar.gz"
-  sha256 "f6f4910fd033078738bd82bfba4f49219d03b17eb0794eb91efbae419f4aba10"
+  url "https://downloads.sourceforge.net/project/lzmautils/xz-5.2.5.tar.xz"
```
Comment (carlocab) [inline Formula/xz.rb:6]: This makes `xz` a build dependency of itself, which probably isn't a good idea.

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/curl.rb (hunk 1 above)
Comment (mhmdanas) [reply Formula/curl.rb:6]: Running `brew info` on the `curl` formula with the `.xz` tarball doesn't mention an `xz` dependency anywhere. Am I missing something?

Comment (SMillerDev) [reply Formula/curl.rb:6]: Yes, brew will download xz if it's needed to unpack a source.

Comment (mhmdanas) [reply Formula/curl.rb:6]: Alright, I'll remove all `.xz` usages I introduced in this PR.

**Comment by mhmdanas** [comment]
How about now? I removed all the `.xz` usages you commented on. Or do I need to remove the other `.xz` tarballs I introduced too?

File: Formula/curl.rb [hunk 5]
```diff
@@ -4,7 +4,6 @@ class Curl < Formula
   url "https://curl.se/download/curl-7.83.1.tar.bz2"
   mirror "https://github.com/curl/curl/releases/download/curl-7_83_1/curl-7.83.1.tar.bz2"
   mirror "http://fresh-center.net/linux/www/curl-7.83.1.tar.bz2"
-  mirror "http://fresh-center.net/linux/www/legacy/curl-7.83.1.tar.bz2"
```
Comment (Bo98) [inline Formula/curl.rb:7]: This mirror becomes valid during the period of time between the next version being released and we update the formula for the new version.

For example:
http://fresh-center.net/linux/www/curl-7.83.0.tar.bz2 is gone
http://fresh-center.net/linux/www/legacy/curl-7.83.0.tar.bz2 is active

**Comment by mhmdanas** [comment]
Is there anything else blocking this PR?

**Pull request closed without merging**